src/did-provider.ts

```diff
@@ -39,35 +40,88 @@ interface DecryptJWEParams {
 
 interface AuthParams {
   paths: Array<string>
+  nonce: string
+  aud?: string
+}
+
+interface JWSSignature {
+  protected: string
+  signature: string
+}
+
+interface GeneralJWS {
+  payload: string
+  signatures: Array<JWSSignature>
+}
+
+function toGeneralJWS(jws: string): GeneralJWS {
+  const [protectedHeader, payload, signature] = jws.split('.')
+  return {
+    payload,
+    signatures: [{ protected: protectedHeader, signature }],
+  }
+}
+
+async function sign(
```

Comment:
Maybe rather than using 6 arguments these could be set in one object to be more explicit when calling the function?

Reply:
yeah, but then I would need to create a new type 🤷